images/origin/scripts/recycler.sh

```diff
@@ -4,40 +4,31 @@
 # this script is intended to be used in a pod that performs the scrub.
 # the container in the pod should succeed or fail based on the exit status of this script.
 
-set -o errexit
 set -o nounset
 set -o pipefail
 
-function recycle(){
+# first and only arg is the directory to scrub
+dir=${1:-}
 
-    # first and only arg is the directory to scrub
-    dir=$1
-
-    if [ -z "$dir" ]; then
-        echo "Usage:  scrub_directory some/path/to/scrub"
-        return 1
-    fi
+if [ -z "${dir}" ]; then
+    echo "Usage:  recycler.sh some/path/to/scrub"
+    exit 1
+fi
 
-    if [ ! -e $dir ]; then
-        echo "scrub directory $dir does not exist"
-        return 1
-    fi
+echo "Scrubbing $dir"
 
-    # remove everything but keep the directory for re-use as a volume
-    rm -rfv $dir/*
+# retain the directory for re-use by another pod.
+# only delete the volume's contents, including dotfiles
+rm -rfv $dir/* $dir/.*
```

Comment:
How about
```bash
shopt -s dotglob nullglob
rm -rf ${dir}/*
```
?


For the current code we have there, if there's no dot file in `$dir`, here's what we get:

```bash
$ mkdir b
$ rm -rf b/.*
rm: refusing to remove ‘.’ or ‘..’ directory: skipping ‘b/.’
rm: refusing to remove ‘.’ or ‘..’ directory: skipping ‘b/..’
$ echo $?
1
```